server/src/main/java/com/cloud/storage/VolumeApiServiceImpl.java

```diff
@@ -2397,6 +2398,35 @@ private void checkDeviceId(Long deviceId, VolumeInfo volumeToAttach, UserVmVO vm
         return volumeToAttach;
     }
 
+    protected void validateIfVmHasBackups(UserVmVO vm, boolean attach) {
+        if ((vm.getBackupOfferingId() == null || CollectionUtils.isEmpty(vm.getBackupVolumeList())) || BooleanUtils.isTrue(BackupManager.BackupEnableAttachDetachVolumes.value())) {
+            return;
+        }
+        String errorMsg = "Unable to detach volume, cannot detach volume from a VM that has backups. First remove the VM from the backup offering or "
+                + "set the global configuration 'backup.enable.attach.detach.of.volumes' to true.";
+        if (attach)
+            errorMsg = "Unable to attach volume, please specify a VM that does not have any backups or set the global configuration "
+                    + "'backup.enable.attach.detach.of.volumes' to true.";
```

Comment:
We can use `BackupManager.BackupEnableAttachDetachVolumes.key()` instead of writing the config name again.